src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/modifier/redundantmodifier/InputRedundantModifierRecords.java

```diff
@@ -0,0 +1,50 @@
+/*
+RedundantModifier
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.modifier.redundantmodifier;
+
+//non-compiled with javac: Compilable with Java16
+public class InputRedundantModifierRecords {
+
+    static record testRecord(int a) { // violation 'Redundant 'static' modifier'
+    }
+
+    interface foo {
+        final static record testRecords(int a) { // 2 violations
+        }
+    }
+
+    class b {
+        static record testRecord(int a) { // violation 'Redundant 'static' modifier'
+        }
+    }
+
+    enum testEnum {
+        ONE;
+        final static record testRecord() { // 2 violations
+
+        }
+
+        class b {
+            static record testRecord() { // violation 'Redundant 'static' modifier'
+            }
+        }
+    }
+}
+
+final record testRecord(int a) { // violation 'Redundant 'final' modifier'
```

Comment:
Add one more top level record, that has no violations as example what is good form of  code.
Our inputs should be combination of valid and invalid input code.

Reply:
done.